### ISSUE TITLE
Filters don't show up anymore

### ISSUE THREAD

**Issue by Pri1133d** [ojchase/OKCupidQuestionFilterExtension#62]
The filters don't show up underneath the Agree/DIsagree/Find out buttons. Think this started happening about 2 weeks ago.

**Comment by ojchase** [comment]
Thanks for the report. (Side note: Wow, this has a real-world user!) OKCupid probably changed the HTML elements on the page that we're waiting to have load or to insert into. Probably not a terribly hard fix.

But I have no capacity for the next five weeks or so. PRs are welcome if you have any technical background. Otherwise I'll get to it when I can. Sorry for the inconvenience!

**Comment by Pri1133d** [comment]
Yes, the extension makes it pretty easy to quickly get a good idea of someone. And I noticed that often the questions give a better (or just different?) impression of a person compared to their profile.

Anyway, JS is not really my thing, but I did have a quick look around. I still see `<div class="profile-questions-filters">`, which seems to be the thing you are currently looking for, but that is as far as I checked.

Haven't even reinstalled the extension because I am not sure how Firefox handles data for extensions and going though all questions again is a pain. (I know I should add an issue for this, but would it be possible to start a filter by saying 'no' to all, and add questions back in?)

**Comment by ojchase** [comment]
Well I'm glad it's been useful to you! It has been to me as well, though I'm not using it right this second while moving across country. 

I probably know even less JS than you. I taught myself for this extension. I'm sure I'm violating all kinds of best practices.

You definitely don't want to reinstall. I'm pretty sure you will lose your data and without a solution to !54 I don't even have a non-manual way to export it. 

As for the bug itself, the console log has an error:
```
TypeError: MutationObserver.observe: Argument 1 is not an object.
```
at
```
[filter.js:209:1]
```

Setting a breakpoint there, `div.profile-questions-filters` doesn't exist _yet_, leading to a null argument which explodes. It looks like `waitForPageToLoad()` is returning too early. Indeed, with the page loading paused at the breakpoint, the profile questions don't exist. There is also no loading div that I can see, so it makes sense that `waitForPageToLoad('.page-loading, .isLoading')` is returning true, as it's waiting for both of those divs to be missing, which they are.

I don't see an obvious quick solution. We need the page to be loaded before we manipulate it, and a programmatic way to detect that isn't jumping out at me yet.

**Comment by Pri1133d** [comment]
Won't you be able to poll it by running a loop that checks every second if it exists?

**Comment by Pri1133d** [comment]
I've been looking at the debugger for a bit, but somehow `filters.js` does not show up. I only see `background_scripts/question_config.js` and `lib/jquery-3.5.1.js`. This also happens after reinstalling the extension and running a local version straight from the repo.

**Comment by ojchase** [comment]
I noticed that too. But the browser console did still show the failing point in `filters.js` with a line number, and clicking that allowed me to set a breakpoint there, which did trigger. My working theory is that the file doesn't stay loaded if it's not actively executing/waiting, and the exception messes that up. And yes, a poll would probably do the job. Doesn't feel the most elegant though.
